parquet-avro/src/main/java/org/apache/parquet/avro/AvroIndexedRecordConverter.java

```diff
@@ -356,12 +356,16 @@ public void end() {
      * @param elementSchema the expected Schema for list elements
      * @return {@code true} if the repeatedType is the element schema
      */
-    static boolean isElementType(Type repeatedType, Schema elementSchema) {
+    static boolean isElementType(Type repeatedType, Schema elementSchema, String parentName) {
       if (repeatedType.isPrimitive() ||
           repeatedType.asGroupType().getFieldCount() > 1) {
         // The repeated type must be the element type because it is an invalid
         // synthetic wrapper (must be a group with one field).
         return true;
+      } else if (repeatedType.asGroupType().getFieldCount() == 1 &&
+          (repeatedType.getName().equals("array") ||
```

Comment:
Rather than checking for "array" here, I think the right thing to do is to check whether the inner type is repeated. That's another case where the structure *must* be a 2-level list because only optional and repeated are allowed there in the 3-level structure.

That check avoids using names, which can be overridden by the requested Avro schema. It will work for both lists of lists and lists of maps.